privacyidea/lib/user.py

```diff
@@ -112,6 +113,10 @@ def _get_user_from_userstore(self):
                 raise UserError("The resolver '{0!s}' does not exist!".format(
                     self.resolver))
             self.uid = y.getUserId(self.login)
+            if y.support_multiple_loginnames:
+                # In case the used loginname was not the primary loginname!
+                self.used_login = self.login
```

Comment:
I'm currently not sure how this will interact with the user cache because it doesn't know about ``used_login``. I guess we would need to test that.

Reply:
I tested it and it shouldn't be a big problem:
Suppose the user cache is enabled and a user authenticates with the secondary login name attribute. This will add a user cache entry, but with the *primary* login name attribute value!
Suppose the user authenticates with the secondary login name attribute again. The user cache will not find an entry and invoke the resolver again to find the user.

... so this might be a bit inefficient because the resolver is asked again for the second authentication request, but the audit log links will work correctly.

Reply:
done in faccf140

Reply:
:+1: